src/H5FDsubfiling/H5FDioc.c

```diff
@@ -0,0 +1,1778 @@
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
```

Comment:
This VFD is the reference I/O concentrator implementation. It works by creating a "main" thread on each MPI rank that is designated as an I/O concentrator rank, as well as a thread pool with a configurable number of worker threads on each of those ranks. "Main" threads will await incoming MPI messages containing I/O requests and will farm those out to the worker threads. The I/O concentrator ranks currently use the sec2 VFD for the underlying subfile, but it should be fairly easy to use other VFDs for file I/O.